code/test/util.js

```diff
@@ -215,6 +269,123 @@ const getEOTP = async ({ seed, hseed, effectiveTime, timeOffset }) => {
   return { index, eotp }
 }
 
+// transactionExecute commits and reveals a wallet transaction
+const transactionExecute = async ({ wallet, operationType, tokenType, contractAddress, tokenId, dest, amount, data, address, randomSeed, backlinkAddresses, testTime }) => {
+  if (testTime === undefined) { testTime = Date.now() }
+  // // calculate counter from testTime
+  const counter = Math.floor(testTime / INTERVAL)
+  const otp = ONEUtil.genOTP({ seed: wallet.seed, counter })
+  // // calculate wallets effectiveTime (creation time) from t0
+  const info = await wallet.wallet.getInfo()
+  const t0 = new BN(info[3]).toNumber()
+  const walletEffectiveTime = t0 * INTERVAL
+  const index = ONEUtil.timeToIndex({ effectiveTime: walletEffectiveTime, time: testTime })
+  const eotp = await ONE.computeEOTP({ otp, hseed: wallet.hseed })
+
+  // Format commit and revealParams based on tokenType
+  let commitParams
+  let revealParams
+  let paramsHash
+  console.log(`operationType: ${operationType}`)
+  switch (operationType) {
+    case ONEConstants.OperationType.TRACK:
+    case ONEConstants.OperationType.UNTRACK:
+      paramsHash = ONEWallet.computeGeneralOperationHash
+      commitParams = { operationType, tokenType, contractAddress, dest }
+      revealParams = { operationType, tokenType, contractAddress, dest }
+      break
+    case ONEConstants.OperationType.OVERRIDE_TRACK:
+      paramsHash = ONEWallet.computeGeneralOperationHash
+      commitParams = { operationType, tokenType, contractAddress, tokenId, dest, amount, data }
+      revealParams = { operationType, tokenType, contractAddress, tokenId, dest, amount, data }
+      break
+    case ONEConstants.OperationType.RECOVER_SELECTED_TOKENS:
+    case ONEConstants.OperationType.BATCH:
+      paramsHash = ONEWallet.computeDataHash
+      commitParams = { operationType, data }
+      revealParams = { operationType, data }
+      break
+    case ONEConstants.OperationType.BACKLINK_ADD:
+    case ONEConstants.OperationType.BACKLINK_DELETE:
+    case ONEConstants.OperationType.BACKLINK_OVERRIDE:
+      paramsHash = ONEWallet.computeDataHash
+      commitParams = { operationType, backlinkAddresses, data }
+      revealParams = { operationType, backlinkAddresses, data }
+      break
+    case ONEConstants.OperationType.COMMAND:
```

Comment:
this is general operation

Reply:
If I understand this correctly 
`ONEWallet.sol` calls `Executor.sol` using
`            Executor.execute(op, tokenTrackerState, backlinkAddresses, signatures, spendingState);`
but only receives in the OperationParameters in the reveal call
**The rest come from the ONEWallets state**
e.g.
`    IONEWallet[] backlinkAddresses; // to be set in next version - these are addresses forwarding funds and tokens to this contract AND must have their forwardAddress updated if this contract's forwardAddress is set or updated. One example of such an address is a previous version of the wallet "upgrading" to a new version. See more at https://github.com/polymorpher/one-wallet/issues/78`
So we should only ever pass Operation Paramaeters 
Even in `commitParams and `revealParams`

Reply:
The type of operation (i.e. how paramsHash should be computed) is determined by https://github.com/polymorpher/one-wallet/blob/477e6f043e3c956ffe8ee921652575e1630feb26/code/contracts/Reveal.sol#L39 , not what is passed to  executor